File: c19_matrix/L240-m.cpp
//
// Created by 李恒 on 2020/7/1.
//
#include <iostream>
#include <vector>

using namespace std;
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if (matrix.size() == 0)
            return false;
        int r = 0;
        int c = matrix[0].size()-1;
        while(r < matrix.size() && c >= 0){
            if (target < matrix[r][c])
                c--;
            else if(target > matrix[r][c])
                r++;
            else
                return true;
        }
        return false;
    }
};

File: python-algorithm/ds_8_matrix/L240-m.py
# -*- coding: utf-8 -*-
# @Time    : 2020/10/4 9:58
# @Author  : No Name

class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if matrix == []:
            return False
        m,n = len(matrix), len(matrix[0])
        i, j = 0, n-1
        while i<m and j>=0:
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] > target:
                j -= 1
            else:
                i += 1
        return False
